can/interfaces/ixxat/canlib_vcinpl2.py

```diff
@@ -0,0 +1,920 @@
+        self._control_handle = HANDLE()
+        self._channel_handle = HANDLE()
+        self._channel_capabilities = structures.CANCAPABILITIES2()
+        self._message = structures.CANMSG2()
+        self._payload = (ctypes.c_byte * 64)()
+
+        # Search for supplied device
+        if UniqueHardwareId is None:
+            log.info("Searching for first available device")
+        else:
+            log.info("Searching for unique HW ID %s", UniqueHardwareId)
+        _canlib.vciEnumDeviceOpen(ctypes.byref(self._device_handle))
+        while True:
+            try:
+                _canlib.vciEnumDeviceNext(
+                    self._device_handle, ctypes.byref(self._device_info)
+                )
+            except StopIteration:
+                if UniqueHardwareId is None:
+                    raise VCIDeviceNotFoundError(
+                        "No IXXAT device(s) connected or device(s) in use by other process(es)."
+                    )
+                else:
+                    raise VCIDeviceNotFoundError(
+                        "Unique HW ID {} not connected or not available.".format(
+                            UniqueHardwareId
+                        )
+                    )
+            else:
+                if (UniqueHardwareId is None) or (
+                    self._device_info.UniqueHardwareId.AsChar
+                    == bytes(UniqueHardwareId, "ascii")
+                ):
+                    break
+                else:
+                    log.debug(
+                        "Ignoring IXXAT with hardware id '%s'.",
+                        self._device_info.UniqueHardwareId.AsChar.decode("ascii"),
+                    )
+        _canlib.vciEnumDeviceClose(self._device_handle)
+
+        try:
+            _canlib.vciDeviceOpen(
+                ctypes.byref(self._device_info.VciObjectId),
+                ctypes.byref(self._device_handle),
+            )
+        except Exception as exception:
+            raise CanInitializationError(f"Could not open device: {exception}")
+
+        log.info("Using unique HW ID %s", self._device_info.UniqueHardwareId.AsChar)
+
+        log.info(
+            "Initializing channel %d in shared mode, %d rx buffers, %d tx buffers",
+            channel,
+            rxFifoSize,
+            txFifoSize,
+        )
+
+        try:
+            _canlib.canChannelOpen(
+                self._device_handle,
+                channel,
+                constants.FALSE,
+                ctypes.byref(self._channel_handle),
+            )
+        except Exception as exception:
+            raise CanInitializationError(
+                f"Could not open and initialize channel: {exception}"
+            )
+
+        # Signal TX/RX events when at least one frame has been handled
+        _canlib.canChannelInitialize(
+            self._channel_handle,
+            rxFifoSize,
+            1,
+            txFifoSize,
+            1,
+            0,
```

Comment:
I'm not aware about message filtering support in python-can, are you talking about implementing some interface (common for all interfaces), so that we enable hardware optimizations for set_filters function?
I could imitate the initialization from yout pull erquest, but I'm not sure about what is better. If you have a strong opinion, I can change it.
We can also leave it for another pull request in next iterations.

Reply:
I agree with your proposal to leave it for another iteration, as I don't really have a strong opinion on that one. Thank you.